src/main/java/org/prebid/server/auction/BidResponseCreator.java

```diff
@@ -89,9 +91,11 @@
     private final String cachePath;
     private final String cacheAssetUrlTemplate;
     private final StoredRequestProcessor storedRequestProcessor;
+    private final Consumer<BidResponse> overwriteBidIdConsumer;
 
     public BidResponseCreator(CacheService cacheService, BidderCatalog bidderCatalog, EventsService eventsService,
-                              StoredRequestProcessor storedRequestProcessor, JacksonMapper mapper) {
+                              StoredRequestProcessor storedRequestProcessor,
+                              JacksonMapper mapper, boolean generateBidId) {
```

Comment:
Can produce NPE. 
Replace to Boolean and use BooleanUtils.isTrue and similar methods